lldb/source/Plugins/Process/elf-core/ProcessElfCore.cpp

```diff
@@ -983,6 +999,40 @@ llvm::Error ProcessElfCore::ParseThreadContextsFromNoteSegment(
   }
 }
 
+bool ProcessElfCore::IsElf(const NT_FILE_Entry entry) {
+  size_t size = strlen(llvm::ELF::ElfMagic);
+  uint8_t buf[size];
```

Comment:
```
15: error: variable length arrays in C++ are a Clang extension [-Werror,-Wvla-cxx-extension]
 1004 |   uint8_t buf[size];
```

Reply:
We can allocate a buffer of 4 and read it.
Can ELFHeader::MagicBytesMatch be used here?